applications/FluidDynamicsApplication/custom_processes/two_fluid_navier_stokes_fractional_convection_process.h

```diff
@@ -0,0 +1,646 @@
+
+        /// Destructor.
+        ~TwoFluidNavierStokesFractionalConvectionProcess() override
+        {
+            mrModel.DeleteModelPart(mAuxModelPartName);
+        }
+
+        ///@}
+        ///@name Operators
+        ///@{
+
+        void operator()()
+        {
+            Execute();
+        }
+
+        ///@}
+        ///@name Operations
+        ///@{
+
+        /**
+         * @brief Perform Navier Stokes Fractional Vectorial Convection.
+         * This solver provides a stabilized convection problem for the fractional velocity unknown.
+         * This process is always called within the framework of the two-fluid Navier-Stokes fractional solver, where the Navier-Stokes equations for two fluids are solved using a splitting approach.
+         * The first part of the splitting involves solving the convection of the fractional velocity 
+         */
+
+
+        void Execute() override
+        {
+            KRATOS_TRY;
+
+            // Fill the auxiliary convection model part if not done yet
+            if (mFractionalVelocityPartIsInitialized == false)
+            {
+                ReGenerateNSFractionalConvectionModelPart(mrBaseModelPart);
+            }
+
+            // If required, calculate nodal element size
+            // Note that this is done once assuming no mesh deformation
+            if (mElementTauNodal || mCalculateNodalH)
+            {
+                ComputeNodalH();
+                mCalculateNodalH = false;
+            }
+
+            // We set these values at every time step as other processes/solvers also use them
+            // Note that this function sets the element related data (e.g. stabilization parameters)
+            auto fill_process_info_function = GetFillProcessInfoFormulationDataFunction();
+            fill_process_info_function(mrBaseModelPart);
+
+            // Set convection problem data
+            auto &r_conv_process_info = mpFractionalVelocityModelPart->GetProcessInfo();
+            const double previous_delta_time = r_conv_process_info.GetValue(DELTA_TIME);
+            const double dt = previous_delta_time;
+
+            // If the nodal stabilization tau is to be used, it is also computed in here
+            IndexPartition<int>(mpFractionalVelocityModelPart->NumberOfNodes()).for_each([&](int i_node)
+                                                                               {
+            const auto it_node = mpFractionalVelocityModelPart->NodesBegin() + i_node;
+            mVelocity[i_node] = it_node->FastGetSolutionStepValue(FRACTIONAL_VELOCITY);
+
+            if (mElementTauNodal) {
+                double velocity_norm = norm_2(mVelocity[i_node]);
+                const double nodal_h = it_node-> GetValue(NODAL_H);
+                const double dynamic_tau = r_conv_process_info.GetValue(DYNAMIC_TAU);
+                const double tau = 1.0 / (dynamic_tau / dt + velocity_norm / std::pow(nodal_h,2));
+
+                it_node->GetValue(TAU) = tau;
```

Comment:
```suggestion
            IndexPartition<int>(mpFractionalVelocityModelPart->NumberOfNodes()).for_each([&](int i_node) {
                const auto it_node = mpFractionalVelocityModelPart->NodesBegin() + i_node;
                mVelocity[i_node] = it_node->FastGetSolutionStepValue(FRACTIONAL_VELOCITY);

                if (mElementTauNodal) {
                    double velocity_norm = norm_2(mVelocity[i_node]);
                    const double nodal_h = it_node-> GetValue(NODAL_H);
                    const double dynamic_tau = r_conv_process_info.GetValue(DYNAMIC_TAU);
                    const double tau = 1.0 / (dynamic_tau / dt + velocity_norm / std::pow(nodal_h,2));

                    it_node->GetValue(TAU) = tau;
```
(minor indentation issue)